### ISSUE TITLE
[Bug]: Use the HorizontalPodAutoscaler v2 for Kubernetes 1.26

### ISSUE THREAD

**Issue by tomrk-esteam8** [jaegertracing/jaeger-operator#2329]
### What happened?

With kubernetes 1.26 HorizontalPodAutoscaler should be in version v2, as beta version is removed. We implement additionaly excluder, that no resources with beta version can be created in the cluster.

Unfortunately operator tries to deploy HorizontalPodAutoscaler with beta version, therefore it does not work at all.

### Steps to reproduce

1. kubernetes 1.26
2. run operator, collector with enabled autoscaling

### Expected behavior

HorizontalPodAutoscaler is created with api version v2

### Jaeger backend version

v 1.49.0

**Comment by pavolloffay** [comment]
We need to support k8s 1.24 (OCP 4.11) 

from: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-25

```
The autoscaling/v2beta1 API version of HorizontalPodAutoscaler is no longer served as of v1.25.

Migrate manifests and API clients to use the autoscaling/v2 API version, available since v1.23.
All existing persisted objects are accessible via the new API
```

So we should be able to bump the version easily.

**Comment by iblancasa** [comment]
Fixed #2374
